[Clang] Warn on backslash-newline-EOF (#97585)
llvm/llvm-project: clang/docs/ReleaseNotes.rst

- Clang now shows implicit deduction guides when diagnosing overload resolution failure. #GH92393.

> - Clang now emits ``-Wnewline-eof`` when the last newline is deleted by a preceding backslash. #GH41571.

Reviewer: ```suggestion
- Clang now emits ``-Wnewline-eof`` when the last newline follows a splicing backslash . #GH41571.
```

Author: Is the space before the full stop intentional?
